### ISSUE TITLE
Bad error reporting on create type

### ISSUE THREAD

**Issue by tailhook** [edgedb/edgedb#1016]
```
CREATE TYPE MyStr EXTENDING str;                                     
ERROR: InternalServerError: 'NoneType' object has no attribute 'is_schema_field'
```
Version `1.0-alpha.2+dev.391.g2276144b`

I obviously needed `CREATE SCALAR TYPE` and would expect it to be grammar error (i.e. something like `TYPE is unexpected, expected one of SCALAR, DATABASE, ...`

**Comment by 1st1** [comment]
Paul, do you think you can look at this one yourself? This needs a relatively simple & quick fix in `edb/schema/` package.

**Comment by tailhook** [comment]
Yes, I can take a look.

**Comment by elprans** [comment]
> would expect it to be grammar error

`CREATE TYPE` is valid syntax and is used to define object types. The crasg in this case is caused by the fact that you're trying to create an object type that inherits from a scalar type.  There has to be a check to prevent mixing schema object classes like that. [Here](https://github.com/edgedb/edgedb/blob/36e86d56cff905b66e865dfb64b0ed3b421ca2ed/edb/schema/inheriting.py#L401) and [here](https://github.com/edgedb/edgedb/blob/36e86d56cff905b66e865dfb64b0ed3b421ca2ed/edb/schema/inheriting.py#L623) are good spots to place the checks in (check `bases`).

**Comment by 1st1** [comment]
@tailhook Please take a look at this one.

**Comment by tailhook** [comment]
Well I've managed to fix exact this error in #1425. I couldn't follow @elprans's hint as execution flow doesn't reach that points.

Wanted to fix similar errors on `ALTER`. But that takes too much time at this moment...

**Comment by msullivan** [comment]
It looks like we produce an error message now (instead of an ISE), but not a good one:
`ERROR: InvalidReferenceError: schema object UUID('00000000-0000-0000-0000-000000000101') exists, but is a 'scalar type' not a 'object type'`

We can do better
